### PR TITLE
feat:  Add IN/OUT Status and Validation for Identical Payee and Payer Addresses

### PR DESCRIPTION
**Pull request by sstefdev**
### Problem
In the Invoice Dashboard, distinguishing between incoming and outgoing transactions is challenging without clear indicators. Additionally, allowing the creation of requests where the payee and payer addresses are identical can lead to logical errors and potential misuse.

### Changes Made
- Added a new transaction status to indicate IN/OUT flow in the Invoice Dashboard.
- Implemented validation to block the creation of requests with identical payee and payer addresses.

#### New tx type
Screenshot 2024-12-17 at 21 31 43

#### New error
Screenshot 2024-12-17 at 17 02 01

## Summary by CodeRabbit

- **New Features**
	- Enhanced invoice form validation to notify users when payer and payee addresses are identical.
	- Added functionality to conditionally display both transaction types when payer and payee are the same in the request view.

- **Bug Fixes**
	- Improved error handling for payer address input, providing clearer user feedback.

- **Documentation**
	- Updated property declarations to reflect new features in the components.

### CODE REVIEW

**Comment by MantisClone** [comment]
Can we make the IN/OUT indicator have diagonal stripes like the Partially Paid indicator?

File: packages/create-invoice-form/src/lib/invoice/form.svelte [hunk 1]
```diff
@@ -338,7 +348,9 @@
             onBlur={checkClientAddress}
             error={validationErrors.clientAddress
               ? "Please enter a valid Ethereum address"
-              : ""}
+              : validationErrors.sameAddress
+                ? "Payer address cannot be the same as recipient address"
```
[the remark this answers is not in the document]
Comment (sstefdev) [reply packages/create-invoice-form/src/lib/invoice/form.svelte:352]: In the UI is called Recipient, but we can add this

**Comment by sstefdev** [comment]
@MantisClone Changes:
- Added diagonal lines to the in/out tx type
- Changed padding so it can be in line with the status

File: packages/create-invoice-form/src/lib/invoice/form.svelte [hunk 2]
```diff
@@ -58,12 +59,21 @@
     validationErrors[`${type}`].email = !isEmail(email);
   };
 
+  const checkSameAddress = () => {
+    return (
+      formData.payerAddress.toLowerCase() ===
+      formData.payeeAddress.toLowerCase()
+    );
+  };
```
[the remark this answers is not in the document]
Comment (rodrigopavezi) [reply packages/create-invoice-form/src/lib/invoice/form.svelte:67]: @sstefdev I think this is valid one

**Comment by MantisClone** [comment]
Can you please update the screenshot in the PR description 🙏

**Comment by sstefdev** [comment]
@MantisClone the screenshots are updated

**Pull request merged**